packages/angular_devkit/build_angular/src/tools/esbuild/bundler-execution-result.ts

```diff
@@ -88,11 +89,15 @@ export class ExecutionResult {
   }
 
   get watchFiles() {
+    // Bundler contexts internally normalize file dependencies
     const files = this.rebuildContexts.flatMap((context) => [...context.watchFiles]);
     if (this.codeBundleCache?.referencedFiles) {
-      files.push(...this.codeBundleCache.referencedFiles);
+      // These files originate from TS/NG and can have POSIX path separators even on Windows.
+      // To ensure path comparisons are valid, all these paths must be normalized.
+      files.push(...this.codeBundleCache.referencedFiles.map(normalize));
```

Comment:
No need to create a copy of the array because map already does that
```js
      files.push(this.codeBundleCache.referencedFiles.map(normalize));
```